src/frontend/js/search-query.ts

```diff
@@ -0,0 +1,106 @@
+        document.getElementById("table-search-suggestions").classList.add("hidden");
+        return;
+    }
+
+    const response = await fetch(url, {
+        method: "POST",
+        headers: {
+            "X-CSRFToken": getCsrfToken(),
+        },
+        body: JSON.stringify({
+            query_string: queryString,
+            object_types: [type],
+        }),
+    });
+
+    if (response.status != 200) {
+        return;
+    }
+
+    const data = await response.json();
+
+    let suggestion_list = document.getElementById("table-search-suggestions");
+    suggestion_list.innerHTML = "";
+    suggestion_list.classList.remove("hidden");
+
+    if (data) {
+        // Set and display new data        
+        for (const value of data.data) {
+            let child = document.createElement("a");
+            child.classList.add(
+                "inline-block",
+                "whitespace-nowrap",
+                "px-4",
+                "py-3",
+                "text-gray-800",
+                "hover:bg-gray-300",
+                "bg-gray-200",
+                "w-full",
+                "overflow-x-hidden",
+                "overflow-ellipsis",
+                "align-top"
+            );
+            child.innerText = value.title;
+            if (value.cms_url != null) {
+                child.setAttribute("href", value.cms_url);
+            }
+            suggestion_list.appendChild(child);
+        }
+    }
+}
+
+let scheduledFunction: number | false = false;
+
+function setSearchQueryEventListeners() {
+    let table_search_input = document.getElementById("table-search-input") as HTMLInputElement;
+
+    // AJAX search
+    table_search_input
+        .addEventListener("keyup", (event) => {
+            event.preventDefault();
+
+            // Reschedule function execution on new input
+            if (scheduledFunction) {
+                clearTimeout(scheduledFunction);
+            }
+            // Schedule function execution
+            scheduledFunction = setTimeout(
+                queryObjects,
+                300,
+                table_search_input.getAttribute("data-url"),
+                table_search_input.getAttribute("data-object-type"),
+                table_search_input.value,
+            );
+        });
+
+    let table_search = document.getElementById("table-search");
+
+    table_search
+        .addEventListener("focusout", (event) => {
+            if (event.relatedTarget && (event.relatedTarget as HTMLElement).closest("#table-search")) {
+                return;
+            }
+            let search_suggestion = document.getElementById("table-search-suggestions");
+            search_suggestion.classList.add("hidden");
+        });
+
+    table_search
+        .addEventListener("focusin", (event) => {
+            let search_suggestion = document.getElementById("table-search-suggestions");
+            search_suggestion.classList.remove("hidden");
+        });
+}
```

Comment:
This should make sense. One thing I am not too sure about is whether this is actually required, since a user can just click search (or press enter) when the specific object they are searching for shows up. So I thought that it might be useful to have a "quick action", but I understand it it might be too much.